src/FakeItEasy/SelfInitializedFakes/MethodInfoConverter.cs

```diff
@@ -0,0 +1,118 @@
+            Debug.Assert(
+                (string)reader.Value == nameof(MethodInfoData.QualifiedTypename),
+                "Unexpected Json property name {0}. Expected: {1}.",
+                (string)reader.Value,
+                nameof(MethodInfoData.QualifiedTypename));
+            reader.Read();
+            var typeName = (string)reader.Value;
+            reader.Read();
+            Debug.Assert(
+                (string)reader.Value == nameof(MethodInfoData.MethodName),
+                "Unexpected Json property name {0}. Expected: {1}.",
+                (string)reader.Value,
+                nameof(MethodInfoData.MethodName));
+
+            reader.Read();
+            var methodName = (string)reader.Value;
+            reader.Read();
+            Debug.Assert(
+                (string)reader.Value == nameof(MethodInfoData.ParameterTypes),
+                "Unexpected Json property name {0}. Expected: {1}.",
+                (string)reader.Value,
+                nameof(MethodInfoData.ParameterTypes));
+            reader.Read();
+            var parameterTypeNames = ((string)reader.Value)?.Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries);
+
+            // Need to advance the file read pointer past the end of current object
+            reader.Read(); // EndArray
+            reader.Read(); // EndObject
+
+            var type = Type.GetType(typeName);
+            var parameterTypes = parameterTypeNames == null ? new Type[0] : parameterTypeNames.Select(name => Type.GetType(name)).ToArray();
+            return type.GetMethod(methodName, parameterTypes);
+        }
+
+        /// <summary>
+        /// Writes the JSON representation of the object.
+        /// </summary>
+        /// <param name="writer">The <see cref="Newtonsoft.Json.JsonWriter"/> to write to.</param>
+        /// <param name="value">The value.</param>
+        /// <param name="serializer">The calling serializer.</param>
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            var methodInfo = (MethodInfo)value;
+            MethodInfoData data = new MethodInfoData
+            {
+                QualifiedTypename = methodInfo.DeclaringType.AssemblyQualifiedName,
+                MethodName = methodInfo.Name,
+                ParameterTypes = methodInfo.GetParameters().Select(p => p.ParameterType.AssemblyQualifiedName).ToArray()
+            };
+            writer.WriteStartObject();
+            writer.WritePropertyName(nameof(data.QualifiedTypename));
+            writer.WriteValue(data.QualifiedTypename);
+            writer.WritePropertyName(nameof(data.MethodName));
+            writer.WriteValue(data.MethodName);
+            writer.WritePropertyName(nameof(data.ParameterTypes));
+            writer.WriteStartArray();
+            for (int i = 0; i < data.ParameterTypes.Length; i++)
+            {
+                writer.WriteValue(data.ParameterTypes[i]);
+            }

```

Comment:
> We've never supported core before, and nothing will change for net40 users.

Well, it won't work if they try to reuse calls that were saved with net40... not sure why they would do that, though (actually, I'm not sure why anyone would want to record and re-read calls, but I assume the use case exists).
Anyway, we can just say that this scenario is not supported.